roles/ovirt-cluster-upgrade/tasks/main.yml

```diff
@@ -0,0 +1,128 @@
+---
+- block:
+  - name: Login to oVirt
+    ovirt_auth:
+      url: "{{ engine_url }}"
+      username: "{{ engine_user }}"
+      password: "{{ engine_password }}"
+      ca_file: "{{ engine_cafile | default(omit) }}"
+      insecure: "{{ engine_insecure | default(true) }}"
+    when: ovirt_auth is undefined
+    register: loggedin
+    tags:
+      - always
+
+  - name: Get hosts
+    ovirt_hosts_facts:
+      auth: "{{ ovirt_auth }}"
+      pattern: "cluster={{ cluster_name | mandatory }} and update_available=true"
```

Comment:
1. There is `upgrade_timeout` variable, which default value is 1200.
2. Not sure I understand. There is no concurency. If using this role, you should use just one target host (usually engine or localhost) to execute it. Concurency would be in this case quite complex, because of migration, so I think we should for now keep it sequantial.